src/thread.rs

```diff
@@ -0,0 +1,12 @@
+use std::io;
+
+use std::thread::{self, JoinHandle};
+
+/// spawn `name`d thread performs `f`
+pub fn spawn_thread<F, R>(name: &str, f: F) -> io::Result<JoinHandle<R>>
+where
+    F: FnOnce() -> R + Send + 'static,
+    R: Send + 'static,
+{
+    thread::Builder::new().name(name.into()).spawn(move || f())
+}
```

Comment:
https://doc.rust-lang.org/src/std/thread/mod.rs.html#596
Just unwrapping the result object seems fine.

Reply:
In the future, the aim is to prevent session errors from affecting the main thread, so escalate this error.